src/UnitTests/GitHub.Collections/Thing.cs

```diff
@@ -0,0 +1,84 @@
+﻿using GitHub.Collections;
+using System;
+using System.ComponentModel;
+
+public class Thing : ICopyable<Thing>, IEquatable<Thing>, IComparable<Thing>, INotifyPropertyChanged
+{
+    public Thing(int id, string title, DateTimeOffset date1, DateTimeOffset date2)
+    {
+        this.Number = id;
+        this.Title = title;
+        this.CreatedAt = date1;
+        this.UpdatedAt = date2;
+    }
+
+    public Thing()
+    {
+    }
+
+    public void CopyFrom(Thing other)
+    {
+        Title = other.Title;
+        CreatedAt = other.CreatedAt;
+        UpdatedAt = other.UpdatedAt;
+    }
+
+    public bool Equals(Thing other)
+    {
+        if (ReferenceEquals(this, other))
+            return true;
+        return other != null && other.Number == Number;
+    }
+
+    public override bool Equals(object obj)
+    {
+        var other = obj as Thing;
+        if (other != null)
+            return Equals(other);
+        return base.Equals(obj);
```

Comment:
I think we can return `false` here because in this case, we know that `obj` is not a `Thing`, but we definitely _are_ a `Thing` so we can't be equal. `base.Equals` in this case tests for reference equality and that can't be true here.